### ISSUE TITLE
Explicitly close connections

### ISSUE THREAD

**Issue by kevinfrommelt-wf** [jtblin/kube2iam#222]
We recently discovered an issue with one of our third-party services running in our K8s cluster. The service was making requests to the AWS metadata server on an interval and each request was making a new connection, but never closing it. TCP keep-alives were being sent by the client and the connections stayed open indefinitely. While I consider this an issue with the third-party service, I noticed there is a difference in behavior between the responses of the AWS metadata server and kube2iam.

The AWS metadata server seems to always respond with the header `Connection: close` indicating the connection should/will be closed.
```
$ curl -v "http://169.254.169.254/2009-04-04/meta-data/instance-id"
*   Trying 169.254.169.254...
* TCP_NODELAY set
* Connected to 169.254.169.254 (169.254.169.254) port 80 (#0)
> GET /2009-04-04/meta-data/instance-id HTTP/1.1
> Host: 169.254.169.254
> User-Agent: curl/7.61.1
> Accept: */*
>
< HTTP/1.1 200 OK
< Content-Type: text/plain
< Accept-Ranges: none
< Last-Modified: Wed, 17 Jul 2019 21:02:29 GMT
< Content-Length: 19
< Date: Thu, 18 Jul 2019 16:09:13 GMT
< Server: EC2ws
< Connection: close
<
* Closing connection 0
<instance_id>
```
```
$ curl -v -H "Connection: keep-alive" -H "Keep-Alive: 60" "http://169.254.169.254/2009-04-04/meta-data/instance-id"
*   Trying 169.254.169.254...
* TCP_NODELAY set
* Connected to 169.254.169.254 (169.254.169.254) port 80 (#0)
> GET /2009-04-04/meta-data/instance-id HTTP/1.1
> Host: 169.254.169.254
> User-Agent: curl/7.61.1
> Accept: */*
> Connection: keep-alive
> Keep-Alive: 60
>
< HTTP/1.1 200 OK
< Content-Type: text/plain
< Accept-Ranges: none
< Last-Modified: Wed, 17 Jul 2019 21:02:29 GMT
< Content-Length: 19
< Date: Thu, 18 Jul 2019 16:10:35 GMT
< Server: EC2ws
< Connection: close
<
* Closing connection 0
<instance_id>
```

kube2iam behaves differently and will respond with the `Connection` header that was given in the request. If the request does not contain the `Connection` header than it will not be in the response.
```
# curl -v "http://169.254.169.254/latest/meta-data/iam/security-credentials/"
* Expire in 0 ms for 6 (transfer 0x55b345efadd0)
*   Trying 169.254.169.254...
* TCP_NODELAY set
* Expire in 200 ms for 4 (transfer 0x55b345efadd0)
* Connected to 169.254.169.254 (169.254.169.254) port 80 (#0)
> GET /latest/meta-data/iam/security-credentials/ HTTP/1.1
> Host: 169.254.169.254
> User-Agent: curl/7.64.0
> Accept: */*
>
< HTTP/1.1 200 OK
< Server: EC2ws
< Date: Thu, 18 Jul 2019 17:45:03 GMT
< Content-Length: 70
< Content-Type: text/plain; charset=utf-8
<
* Connection #0 to host 169.254.169.254 left intact    <-- Connection stays open
<iam_role_arn>
```
```
# curl -v -H "Connection: close" "http://169.254.169.254/latest/meta-data/iam/security-credentials/"
* Expire in 0 ms for 6 (transfer 0x5635d53aedd0)
*   Trying 169.254.169.254...
* TCP_NODELAY set
* Expire in 200 ms for 4 (transfer 0x5635d53aedd0)
* Connected to 169.254.169.254 (169.254.169.254) port 80 (#0)
> GET /latest/meta-data/iam/security-credentials/ HTTP/1.1
> Host: 169.254.169.254
> User-Agent: curl/7.64.0
> Accept: */*
> Connection: close
>
< HTTP/1.1 200 OK
< Server: EC2ws
< Date: Thu, 18 Jul 2019 18:04:35 GMT
< Content-Length: 70
< Content-Type: text/plain; charset=utf-8
< Connection: close
<
* Closing connection 0
<iam_role_arn>
```

So long story short, I'm wondering if it would be better to emulate the behavior of the AWS metadata server and explicitly send `Connection: close` in every response.
